sdk/cosmosdb/cosmos/src/queryExecutionContext/ContinuationTokenManager.ts

```diff
@@ -0,0 +1,801 @@
+import type { OrderByQueryContinuationToken } from "../documents/ContinuationToken/OrderByQueryContinuationToken.js";
+import { 
+  createOrderByQueryContinuationToken,
+  serializeOrderByQueryContinuationToken,
+  parseOrderByQueryContinuationToken
+} from "../documents/ContinuationToken/OrderByQueryContinuationToken.js";
+import type { CosmosHeaders } from "./CosmosHeaders.js";
+import { Constants } from "../common/index.js";
+
+/**
+ * Manages continuation tokens for multi-partition query execution.
+ * Handles composite continuation token creation, range mapping updates, and token serialization.
+ * Supports both parallel queries (multi-range aggregation) and ORDER BY queries (single-range sequential).
+ * @hidden
+ */
+export class ContinuationTokenManager {
+  private compositeContinuationToken: CompositeQueryContinuationToken;
+  private partitionKeyRangeMap: Map<string, QueryRangeMapping> = new Map();
+  private isOrderByQuery: boolean = false;
+  private orderByQueryContinuationToken: OrderByQueryContinuationToken | undefined;
+  private orderByItemsArray: any[][] | undefined;
+
+  constructor(
+    private readonly collectionLink: string,
+    initialContinuationToken?: string,
+    isOrderByQuery: boolean = false,
+  ) {
+    this.isOrderByQuery = isOrderByQuery;
+    if (initialContinuationToken) {
+      try {
+        // Parse existing continuation token for resumption
+        console.log(
+          `Parsing continuation token for ${isOrderByQuery ? "ORDER BY" : "parallel"} query`,
+        );
+
+        if (this.isOrderByQuery) {
+          const parsedToken = JSON.parse(initialContinuationToken);
+          if (parsedToken && parsedToken.compositeToken && parsedToken.orderByItems) {
+            console.log("Detected ORDER BY continuation token with composite token");
+            this.orderByQueryContinuationToken = parsedToken as OrderByQueryContinuationToken;
+
+            // Extract the inner composite token
+            this.compositeContinuationToken = CompositeQueryContinuationTokenClass.fromString(
+              parsedToken.compositeToken,
+            );
+          }
+        } else {
+          // For parallel queries, expect a CompositeQueryContinuationToken directly
+          console.log("Parsing parallel query continuation token as composite token");
+          this.compositeContinuationToken =
+            CompositeQueryContinuationTokenClass.fromString(initialContinuationToken);
+        }
+
+        console.log(
+          `Successfully parsed ${isOrderByQuery ? "ORDER BY" : "parallel"} continuation token`,
+        );
+      } catch (error) {
+        console.warn(
+          `Failed to parse continuation token: ${error.message}, initializing empty token`,
+        );
+        // Fallback to empty continuation token if parsing fails
+        this.compositeContinuationToken = new CompositeQueryContinuationTokenClass(
```

Comment:
Don't fallback. Better to fail the request. 
Implementation should be transparent and deterministic for users.

Reply:
addressed